core/src/main/java/org/elasticsearch/action/get/TransportMultiGetAction.java

```diff
@@ -64,26 +64,21 @@ protected void doExecute(final MultiGetRequest request, final ActionListener<Mul
         for (int i = 0; i < request.items.size(); i++) {
             MultiGetRequest.Item item = request.items.get(i);
 
-            if (!clusterState.metaData().hasConcreteIndex(item.index())) {
-                responses.set(i, newItemFailure(item.index(), item.type(), item.id(), new IndexNotFoundException(item.index())));
-                continue;
-            }
-
             String concreteSingleIndex;
             try {
-                item.routing(clusterState.metaData().resolveIndexRouting(item.parent(), item.routing(), item.index()));
                 concreteSingleIndex = indexNameExpressionResolver.concreteSingleIndex(clusterState, item).getName();
-
-                if ((item.routing() == null) && (clusterState.getMetaData().routingRequired(concreteSingleIndex, item.type()))) {
-                    String message = "routing is required for [" + concreteSingleIndex + "]/[" + item.type() + "]/[" + item.id() + "]";
-                    responses.set(i, newItemFailure(concreteSingleIndex, item.type(), item.id(), new IllegalArgumentException(message)));
-                    continue;
-                }
             } catch (Exception e) {
                 responses.set(i, newItemFailure(item.index(), item.type(), item.id(), e));
                 continue;
             }
 
+            item.routing(clusterState.metaData().resolveIndexRouting(item.parent(), item.routing(), concreteSingleIndex));
+            if ((item.routing() == null) && (clusterState.getMetaData().routingRequired(concreteSingleIndex, item.type()))) {
+                String message = "routing is required for [" + concreteSingleIndex + "]/[" + item.type() + "]/[" + item.id() + "]";
+                responses.set(i, newItemFailure(concreteSingleIndex, item.type(), item.id(), new IllegalArgumentException(message)));
+                continue;
+            }
```

Comment:
Got it, just pushed the change. I think that exception in the routing code is checking for a multi index alias, so that should be covered in the test `SimpleMgetIT#testThatMgetShouldWorkWithMultiIndexAlias`. However I'm pretty sure it wont actually reach that code since `indexNameExpressionResolver` will throw an exception before.
Thanks for the review!